python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -266,8 +287,8 @@ def _parse_op_fn(op: AbstractUDFMap):
             op_fn = make_callable_class_concurrent(op_fn)
 
         def init_fn():
-            if ray.data._map_actor_context is None:
-                ray.data._map_actor_context = _MapActorContext(
+            if wrapper.map_actor_context is None:
+                wrapper.map_actor_context = _MapActorContext(
```

Comment:
this function will be called in a different process. 
If we simply capture the `wrapper`. `init_fn` and `fn` will capture 2 different instances of the wrappers. 
We need to pass the wrapper class to the map task. And `init_fn/fn` need to be defined in the wrapper and access the context with `self.map_actor_context`.